Added Firmware Documentation

davidetome opened the pull request:
This PR ports and updates the old wiki documentation (in terms of updating links and adding some resources).
In particular, the involved wiki pages are :

- https://wiki.icub.org/wiki/Firmware
- https://wiki.icub.org/wiki/CanLoader
- https://wiki.icub.org/wiki/FT_sensor
- https://wiki.icub.org/wiki/Can_addresses_and_associated_firmware
- https://redmine.robotology.eu/projects/hardware/wiki/Force-Torque_sensors

⚠️ The documentation structure should be discussed and modified if necessary. Please feel free to propose any changes, additions or fixes

cc @pattacini @traversaro @maggia80

pattacini commented on mkdocs.yml:
```diff
@@ -115,3 +115,9 @@ nav:
+        - Firmware Documentation: icub_firmware/firmware/firmware.md
+        - Associated Firmware: icub_firmware/associated-firmware/associated-firmware.md
+        - CanLoader Tools: icub_firmware/canLoader/canLoader.md
+        - F/T Sesnors: icub_firmware/ft-sensor/ft-sensor.md
```
```suggestion
        - F/T Sensors: icub_firmware/ft-sensor/ft-sensor.md
```

pattacini commented on docs/icub_firmware/ft-sensor/ft-sensor.md:
```diff
@@ -0,0 +1,269 @@
+
+## Resources
+
+- [FT58 - without Temp. Compensation](./assets/FT58%20-%20without%20Temp.%20Compensation.pdf)
+- [FT45 - Temp. Compensation included](./assets/FT45%20-%20Temp.%20Compensation%20included.PDF)
+- [FT45 - without Temp. Compensation](./assets/FT45%20-%20without%20Temp.%20Compensation.PDF)
+- [FT45 - Temp. Compensation included 2](assets/FT45%20-%20Temp.%20Compensation%20included-2.PDF)
+- [FT58 - Temp. Compensation included](assets/FT58%20-%20Temp.%20Compensation%20included.PDF)
```
```suggestion
- [FT45 - Temp. Compensation included](./assets/FT45%20-%20Temp.%20Compensation%20included.PDF)
- [FT45 - without Temp. Compensation](./assets/FT45%20-%20without%20Temp.%20Compensation.PDF)
- [FT45 - Temp. Compensation included 2](assets/FT45%20-%20Temp.%20Compensation%20included-2.PDF)
- [FT58 - Temp. Compensation included](assets/FT58%20-%20Temp.%20Compensation%20included.PDF)
- [FT58 - without Temp. Compensation](./assets/FT58%20-%20without%20Temp.%20Compensation.pdf)
```

traversaro commented:
I am not sure about the canLoader section, if I recall correctly now everything is handled by the FirmwareUpdater, right?

traversaro commented:
This docs may be interesting for: @isorrentino @GiulioRomualdi @giovannipizzolante @S-Dafarra

davidetome commented:
> I am not sure about the canLoader section, if I recall correctly now everything is handled by the FirmwareUpdater, right?

you're right but I ported the doc as is seen the fact it speaks about both `FirmwareUpdater` and `canLoader`.. May be for old fashioned users still be useful?

pattacini commented:
@traversaro, @davidetome and I have just discussed `FirmwareUpdater` vs. `CanLoader` and we opted for amending the documentation in order to:
- give priority to FirmwareUpdater and ask @marcoaccame whether there are more recent info on this;
- keep the documentation of the CanLoader separately as it could be still a useful resource in some cornercases. 

How do you see that?

traversaro commented:
> keep the documentation of the CanLoader separately as it could be still a useful resource in some corner-cases.

We can definitely cleanup this later, no problem on my side. But just to clarify: the canLoader GUI tool has been removed from icub-main in https://github.com/robotology/icub-main/pull/659, so unless the users are using the quite old distro 2020.02, so I don't think it is useful for any user to have docs on it.

davidetome commented:
I've just done some cleanup, the main actions are 👍🏻 

- Moved all info about `CanLoader` in the dedicated page adding the info that's deprecated
- Some cleanup on explicit links
-  Updated associated firmware page

cc @pattacini @traversaro

pattacini commented on docs/icub_firmware/ft-sensors/ft-sensors.md:
```diff
@@ -0,0 +1,269 @@
+
+## Resources
+
+- [FT58 - without Temp. Compensation](./assets/FT58%20-%20without%20Temp.%20Compensation.pdf)
+- [FT45 - Temp. Compensation included](./assets/FT45%20-%20Temp.%20Compensation%20included.PDF)
+- [FT45 - without Temp. Compensation](./assets/FT45%20-%20without%20Temp.%20Compensation.PDF)
+- [FT45 - Temp. Compensation included 2](assets/FT45%20-%20Temp.%20Compensation%20included-2.PDF)
+- [FT58 - Temp. Compensation included](assets/FT58%20-%20Temp.%20Compensation%20included.PDF)
```
```suggestion
- [FT45 - without Temp. Compensation](./assets/FT45%20-%20without%20Temp.%20Compensation.PDF)
- [FT45 - Temp. Compensation included](./assets/FT45%20-%20Temp.%20Compensation%20included.PDF)
- [FT45 - Temp. Compensation included 2](./assets/FT45%20-%20Temp.%20Compensation%20included-2.PDF)
- [FT58 - without Temp. Compensation](./assets/FT58%20-%20without%20Temp.%20Compensation.pdf)
- [FT58 - Temp. Compensation included](./assets/FT58%20-%20Temp.%20Compensation%20included.PDF)
```

pattacini reviewed (approved):
Nicely done @davidetome 🚀 

I'll be putting some redirections within the migrated pages.

Concerning the FirmwareUpdater, I've realized that the documentation is very old-style and could be overhauled in the future, @marcoaccame

traversaro commented:
Thanks @davidetome ! A great way to kick-off a 2021 rich of innovations for the iCub firmware. : )
